### PR TITLE
02.Overview/03.Identity: restructure

### PR DESCRIPTION
**Pull request by mchalski**
https://tracker.mender.io/browse/MEN-3183

besides the obvious `passive->active voice` thing - I addressed some stuff that sounded clunky to me, hope it's more friendly and to the point now. 

bonus: top level chapter.md file - important for navigation.

@estenberg I got a few comments already from different folks - I'd appreciate if you gave them a final ack/nack as well. you have the final word on this I guess, and then I could fix and close all of it in one go.

### CODE REVIEW

File: 02.Overview/03.Identity/docs.md [hunk 1]
```diff
@@ -7,8 +7,8 @@ taxonomy:
 
 ##Definition
 
-Formally, the definition of an identity is a set of key-value attributes.
-For example, the identity of a person could be defined as:
+An identity is a set of key-value attributes.
+For example, you could describe a person as:
```
Comment (oleorhagen) [inline 02.Overview/03.Identity/docs.md:11]:
```suggestion
For example, you can describe a person as:
```

File: 02.Overview/03.Identity/docs.md [hunk 2]
```diff
@@ -18,18 +18,18 @@ Address=Jone's way 3, 1234 Boston
...
 can adapt to the identity scheme of any environment. However, Mender imposes the following
 requirements for device identities:
 
 * the combination of the attribute values must be *unique* to each device, so that identities are not ambiguous when deploying software
-* identity attributes must *never change* for the lifetime of a device, so that a device can be recognized over its lifetime
+* identity attributes must *never change* for the lifetime of a device, so that the Mender server can recognize them reliably
 
-Note that device keys should not be used as part of identities, because this would make it very difficult to
-rotate or regenerate keys over the lifetime of the device as it would in effect change identity.
+Do not use device keys as part of an identity, because this will make it very difficult to
```
Comment (oleorhagen) [inline 02.Overview/03.Identity/docs.md:31]:
```suggestion
Do not use device keys as part of an identity, because this makes it very difficult to
```

File: 02.Overview/03.Identity/docs.md [hunk 3]
```diff
@@ -18,18 +18,18 @@ Address=Jone's way 3, 1234 Boston
...
+rotate or regenerate keys over the lifetime of the device (as it would in effect change identity).
```
Comment (oleorhagen) [inline 02.Overview/03.Identity/docs.md:32]:
```suggestion
rotate or regenerate keys over the lifetime of the device (as it in effect changes the identity).
```

File: 02.Overview/03.Identity/docs.md [hunk 4]
```diff
@@ -83,13 +82,13 @@ stored by the Mender server. The Device ID is primarily used for security purpos
 recognition of devices during device authentication, grouping, reporting, and similar cases.
 
 
-##Implementing a device identity executable
+##Implement a device identity executable
```
Comment (oleorhagen) [inline 02.Overview/03.Identity/docs.md:85]: :+1:

File: 02.Overview/03.Identity/docs.md (hunk 1 above)
Comment (mirzak) [inline 02.Overview/03.Identity/docs.md:11]: We should update this example using something more relevant. We are dealing with devices and not people. 

Example:

```
ID={
Customer-Name=Northern Tech AS
Project-ID=4306609
Site=Oslo
Serial-Number=12345678
}
```

File: 02.Overview/03.Identity/docs.md [hunk 5]
```diff
@@ -38,24 +38,23 @@ It is important to have the ability to regenerate keys if a device gets compromi
 Even though Mender allows several attributes to make up the device identity,
 most environments just use one attribute for simplicity.
 
-The MAC address of the primary interface is a common way to satisfy the requirements for device identity,
-for example:
+For example, a MAC address of the primary interface satisfies the requirements for device identity:
```
Comment (mirzak) [inline 02.Overview/03.Identity/docs.md:41]: Would also list eMMC CID as an example.

```
# cat /sys/class/mmc_host/mmc0/mmc0\:0001/cid 
90014a484247346132a4528894a9a300
```

This is kind of the MAC of eMMC flash chips

File: 02.Overview/03.Identity/docs.md [hunk 6]
```diff
@@ -74,7 +73,7 @@ the MAC address to identify others.
 
 For convenience it is very useful to always have a fixed-length string that uniquely identifies a
 device. The Mender server implements this by assigning a unique **Device ID** to a device
-as it is connected to the server.
+as it connects to the server.
 
 For example, a Device ID could look like `594019224d36350001fc5cc3`.
```
Comment (mirzak) [inline 02.Overview/03.Identity/docs.md:78]: Device Id looks like `be979170-388a-4f38-871d-a4cf88b3ba2d` nowadays?

File: 02.Overview/chapter.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
+---
```
Comment (drewmoseley) [inline 02.Overview/chapter.md:1]: Why is this here and part of this PR? Is it intentional?

File: 02.Overview/02.Device-authentication/docs.md [hunk 8]
```diff
@@ -4,3 +4,174 @@ taxonomy:
...
+
+* [Device Authentication Device
+  API](../../apis/enterprise/device-apis/device-authentication)
+* [Device Authentication Management
+  API](../../apis/enterprise/management-apis/device-authentication)
+
+### Preauthorization Flow
+
+Preauthorization is the idea of authorizing a device before it ever connects to
+the server for the first time. This is the intuitive model analogous to creating
```
[the remark this answers is not in the document]
Comment (mchalski) [reply 02.Overview/02.Device-authentication/docs.md]: sorry, bad rebase - this is not part of this PR.
I accidentally pulled an unrelated commit by Lluis, removed now.

File: 02.Overview/02.Device-authentication/docs.md [hunk 9]
```diff
@@ -4,3 +4,174 @@ taxonomy:
...
+an account before logging in to an online service.
+
+It allows to authorize a particular device before is released or even assembled.
```
[the remark this answers is not in the document]
Comment (mchalski) [reply 02.Overview/02.Device-authentication/docs.md]: sorry, bad rebase - this is not part of this PR.
I accidentally pulled an unrelated commit by Lluis, removed now.

File: 02.Overview/02.Device-authentication/docs.md [hunk 10]
```diff
@@ -4,3 +4,174 @@ taxonomy:
...
+It is sufficient for the user to submit a pre-assigned authentication set to
+Device Authentication. At some point a device with corresponding identity
```
[the remark this answers is not in the document]
Comment (mchalski) [reply 02.Overview/02.Device-authentication/docs.md]: sorry, bad rebase - this is not part of this PR.
I accidentally pulled an unrelated commit by Lluis, removed now.

File: 02.Overview/02.Device-authentication/docs.md [hunk 11]
```diff
@@ -4,3 +4,174 @@ taxonomy:
...
+|:--:|
+|*Preauthorization flow*|
+
+### Accept-on-request Flow
+
+An alternate flow, suitable mostly for quick prototyping and testing, is the
+accept-on-request flow.
+
+It is not required for the user to preauthorize a device - instead, when the
+device first submits an auth request gets recorded for future inspection by the
```
[the remark this answers is not in the document]
Comment (mchalski) [reply 02.Overview/02.Device-authentication/docs.md]: sorry, bad rebase - this is not part of this PR.
I accidentally pulled an unrelated commit by Lluis, removed now.

File: 02.Overview/02.Device-authentication/docs.md [hunk 12]
```diff
@@ -4,3 +4,174 @@ taxonomy:
...
+user. The auth set can then be accepted via the Device Authentication API
+whenever the user sees fit; a subsequent auth request from the device will
+result in a successful authorization.
```
[the remark this answers is not in the document]
Comment (mchalski) [reply 02.Overview/02.Device-authentication/docs.md]: sorry, bad rebase - this is not part of this PR.
I accidentally pulled an unrelated commit by Lluis, removed now.

File: 02.Overview/chapter.md (hunk 7 above)
Comment (mchalski) [reply 02.Overview/chapter.md:1]: yes, without chapter files the documentation doesn't render at all.
this means you can't really verify your changes in the browser before pushing a PR.
all of these were removed by Ole during initial restructuring (accident).
we'll be bringing them back one by one as we go.

File: 02.Overview/03.Identity/docs.md (hunk 2 above)
Comment (mchalski) [reply 02.Overview/03.Identity/docs.md]: fixed

File: 02.Overview/03.Identity/docs.md (hunk 1 above)
Comment (mchalski) [reply 02.Overview/03.Identity/docs.md]: fixed

File: 02.Overview/03.Identity/docs.md (hunk 6 above)
Comment (mchalski) [reply 02.Overview/03.Identity/docs.md]: fixed

File: 02.Overview/03.Identity/docs.md (hunk 5 above)
Comment (mchalski) [reply 02.Overview/03.Identity/docs.md:41]: trusting you on this one, fixed

File: 02.Overview/03.Identity/docs.md (hunk 1 above)
Comment (mchalski) [reply 02.Overview/03.Identity/docs.md]: agreed, this sounds too for-dummies.

I pulled your example in.

File: 02.Overview/03.Identity/docs.md [hunk 13]
```diff
@@ -7,29 +7,30 @@ taxonomy:
...
-For example, the identity of a person could be defined as:
+An identity is a set of key-value attributes.
+For example, you can describe a device as:
 
 ```
 ID={
-Name=Tom Jones
-SSN=12345678
-Address=Jone's way 3, 1234 Boston
+CustomerName=Northern Tech AS
```
Comment (mirzak) [inline 02.Overview/03.Identity/docs.md:15]: Sorry, but this should probably be `Northern.tech AS` :).

Comment (mchalski) [reply 02.Overview/03.Identity/docs.md]: good looking out, fixed :)

File: 02.Overview/03.Identity/docs.md [hunk 14]
```diff
@@ -7,29 +7,30 @@ taxonomy:
...
-For example, the identity of a person could be defined as:
+An identity is a set of key-value attributes.
+For example, you can describe a device as:
 
 ```
 ID={
-Name=Tom Jones
-SSN=12345678
-Address=Jone's way 3, 1234 Boston
+CustomerName=Northern.Tech AS
```
Comment (mirzak) [inline 02.Overview/03.Identity/docs.md:15]: One more round on this one :). It should be a lowercase t,

```
CustomerName=Northern.tech AS
```

Comment (mchalski) [reply 02.Overview/03.Identity/docs.md]: ruthless! fixed :)

**Comment by mchalski** [comment]
@drewmoseley I resolved your comments, but need an explicit 'approve' if there's nothing else.

(or from @michaelatmender; this is according to our new `must have a native speaker's approval` rule).
